### ISSUE TITLE
Accessibility: Service Menu contact form without breadcrumb

### ISSUE THREAD

**Issue by CS-Markus** [shopware/shopware#7617]
*PHP Version:* 8.3   | *Shopware Version:* 6.6.10.2  | *Affected area / extension:* Platform(Default) 

 ----

*_Actual behaviour:_*

**EN**    
If ACCESSIBILITY_TWEAKS=1 is active in the .env, no footer menu is displayed when clicking on the service menu link (see screenshot) and no breadcrumb is generated. 

**DE**  
Wenn ACCESSIBILITY_TWEAKS=1 in der .env aktiv ist, wird beim Klick auf den Servicemenu Link (siehe Screenshot) kein Footermenü angezeigt und kein Breadcrumb erzeugt.

![Image](https://github.com/user-attachments/assets/b953d31d-cb63-4fd9-a145-978c6d7e3649)

*_Expected behaviour:_*

The Footer menu and the breadcrumb is shown

*_How to reproduce:_*

Click on the Link in the service menu

**Comment by shopwareBot** [comment]
[public] Shopware automated response: This issue is linked to the internal issue: **NEXT-41153**



[created from NEXT-41153, comment 545401]

**Comment by shopwareBot** [comment]
[public] 

 > Priority changed to: **High**

 Thank you for your report. Based on the information in this report, we have decided to investigate this issue in more detail and are planning to provide a fix for this issue. Although we are always eager to solve such problems as soon as we can, we can't give you a more specific time frame right now. We appreciate your patience and thank you for your understanding.



[created from NEXT-41153, comment 545438]

**Comment by jhit** [comment]
Same problem with the link to shipping costs in `footer-vat` in the text snippet `footer.includeVatText`

**Comment by tobiasberge** [comment]
Hi @CS-Markus @jhit,

thanks for the report. Just to clarify I understand the issue correctly. After clicking the link in the footer, the following page (e.g. contact form or shipping information page) is missing page data. The footer menu items and menu/breadcrumb items are missing?


I can reproduce this in 6.6.10.2 but not anymore in 6.7. We should then remove the 6.7-RC issues label @BrocksiNet. But it is still a valid issue in 6.6.

### 6.6.10.2:

![Image](https://github.com/user-attachments/assets/1c527812-6acc-47d5-806c-d3d033163c3a)

### 6.7.0 / trunk:

(I have not generated the footer menu here, but I can confirm that it is missing in 6.6 on "normal pages" vs when using the footer link)

![Image](https://github.com/user-attachments/assets/39685c41-cff1-4718-a743-9f6ffb512c5e)

**Comment by jhit** [comment]
@tobiasberge wrote:
> Thank you for the report. I would like to clarify that I understand the issue correctly. After clicking the link in the footer, the subsequent page (for example, the contact form or shipping information page) is missing certain data. Are the footer menu items and the menu/breadcrumb items absent as well?

This accurately describes the issue. 
The underlying problem in the code is that the links are generated using the Twig `path()` function with the CMS page controller `frontend.cms.page.full`, instead of using a navigation controller like `frontend.navigation.page`.

Regarding 6.7.0.:
This is a mixed bag of issues. Yes, the navigation shows up. But:

1. The Breadcrumb is missing
2. The URL in the browser address bar is not an SEO URL

You cannot say that 6.7.0 is working as it should. See the attached screenshot.

![Image](https://github.com/user-attachments/assets/ff7a858c-abf1-411c-b02f-dcea69335e85)

And this is how it should look like:

![Image](https://github.com/user-attachments/assets/22e9a98d-1097-45c7-90c0-879b880b6a90)

**Comment by keulinho** [comment]
The issue that the service menu is missing in the contact forms was fixed in 6.7.

However breadcrumbs and seo-urls are not supported out of the box, as those "routes" (contact form / data protections information, ...) are not part of the regular category tree and therefore no SEO-url or breadcrumb information is available for those. 

The workaround currently would be to create regular shop pages with contact forms etc and use those in the service menu in the footer. 

We'll remove this from 6.7 RC milestone. 

Currently we are thinking in the direction of creating those shop pages per default and link those.

**Comment by keulinho** [comment]
One pot. solution could be to allow setting categories in the basic information -> shop pages, instead of or additionally to the direct cms layouts, that way we would have SEO URL and breadcrumb information

**Comment by jhit** [comment]
> One pot. solution could be to allow setting categories in the basic information -> shop pages, instead of or additionally to the direct cms layouts, that way we would have SEO URL and breadcrumb information

Yes, that is exactly how I implemented my workaround in the Theme Plugin.

**Comment by ShopwareM** [comment]
Google Search doesn’t really like the breadcrumbs of empty footer categories.

<img width="744" alt="Image" src="https://github.com/user-attachments/assets/d66460cc-f979-482c-b7ab-f8643413805f" />

**Comment by taltholtmann** [comment]
Just checked the current behavior.

The Footer Menu:
- for 6.6: the menu will be shown again, if the feature flag `CACHE_REWORK=1` is active.
- for 6.7: It works without changing anything

The breadcrumb:
Per default. The contact and payment info are just cms pages without a category. So there is also no breadcrumb generated.
But in both versions(6.6.10.7 and 6.7.x) I could create categories in the footer and just assign these cms pages. Then you have breadcrumbs according to your category structure, here is the payment example:

<img width="2118" height="2594" alt="Image" src="https://github.com/user-attachments/assets/c0f1d556-dee3-4e07-848a-47af82f4b414" />

Now, the snippets for the contact form and vat info can be adjusted to point to these pages.

I will close this ticket for now as I don't see a bug to be solved. We could consider opening a feature request to simplify the steps to get a service/vat page with a breadcrumb.

**Comment by jhit** [comment]
I have another two links that need to be changed in the text snippets:
*  `cookie.messageTextPage`. This one will appear in the cookie consent pop-up.
* `footer.includeVatTextPage`: Contains a link to the payment and shipping page.

There is possibly more.

**Comment by jhit** [comment]
@taltholtmann The "bug" is that you now have to manually maintain every link discussed in this issue for every language used in your shop. From the state where everything "just works", we are now in the state where you have to edit every link manually and also keep track of all the links in different languages for changes as long as the shop exists.

**Comment by taltholtmann** [comment]
@jhit Thank you for the additional information. But what I dont get is, you said "everything just worked" before. Could you explain that? 
To my knowledge, it was the same problem before when you used the categories to create pages with breadcrumbs for these cms pages. Also they did not have breadcrumbs when opened directly.
The thing that was changed because of accessibility, is, that the modals are not used anymore.

**Comment by jhit** [comment]
@alvarothomas Coming from Shopware 6.6 and earlier, all those problematic links were opened in a modal window. I think this is the primary reason why all the Shop Pages configured in Settings > Basic Information > Shop Pages point to Layouts and not Categories. For Modal Windows, Layouts are optimal.

Since Shopware 6.6 and the ACCESSIBILITY_TWEAKS flag, these links no longer open in a Modal but directly link to a regular storefront page. 
New text snippets were created that changed the code from Modal to regular links, like `footer.includeVatText` => `footer.includeVatTextPage`.
This created the problem that linking to a Layout will cause issues because the context where you are (Category) is missing.

In Shopware <= 6.6, all the "Shop Pages" links spread in the storefront would "just work". They would always open the correct Modal window with the content in the proper language. All you needed to do was configure the Layouts in the basic shop information.
If you decided to rename a Layout, it would not have had any consequences because the linking was done by the UUID, and most "Shop Pages" were properly predefined in every new Shop installation. Now I need to create my footer menu first and edit the Link in the text snippets. Each time someone decides to rename a category, I have to correct the corresponding text snippet in all languages.

**Comment by taltholtmann** [comment]
Thanks for your input and discussion around this topic!  
I’ve created a new improvement ticket to address the currently inconvenient behavior and to simplify how shop pages (like contact, payment, etc.) are handled when rendered as full pages instead of modals:  
👉 [#13226](https://github.com/shopware/shopware/issues/13226)

Please feel free to share any additional ideas or possible solution approaches in that new ticket.  
Your feedback has been very helpful — thank you all for your participation!
